ms2rescore/utils.py

```diff
@@ -91,3 +97,125 @@ def _is_minitdf(spectrum_file: str) -> bool:
     files = set(Path(spectrum_file).glob("*ms2spectrum.bin"))
     files.update(Path(spectrum_file).glob("*ms2spectrum.parquet"))
     return len(files) >= 2
+
+
+def _strip_v(s: str) -> str:
+    return s.lstrip("vV").strip() if isinstance(s, str) else s
+
+
+def _normalize_for_tuple(v: str):
+    """
+    Very small-footprint normalizer that yields a tuple suitable for comparison
+    when 'packaging' is unavailable. It prefers numeric precedence and
+    compares leftover text segments lexically. Pre-releases are not modeled,
+    but common tags like 'v1.2.3' vs '1.10.0' work as expected.
+    """
+    v = _strip_v(v)
+    # Split into alternating number / text chunks: "1.2.3-rc1" -> ['1','.', '2','.', '3','-','rc','1']
+    chunks = re.findall(r"\d+|[A-Za-z]+|[^A-Za-z0-9]+", v)
+    norm = []
+    for c in chunks:
+        if c.isdigit():
+            norm.append((0, int(c)))
+        else:
+            # collapse punctuation to a single separator to keep ordering stable
+            txt = c if re.search(r"[A-Za-z]", c) else "."
+            norm.append((1, txt))
+    return tuple(norm)
+
+
+def _compare_versions(a: str, b: str) -> int:
+    """
+    Returns -1 if a<b, 0 if a==b, 1 if a>b.
+    Tries packaging.version.Version if available; otherwise falls back to tuple comparison.
```

Comment:
What is the reason not to just require packaging, or just always use the own parser?